packages/web/components/transactions/transaction-buttons.tsx

```diff
@@ -0,0 +1,107 @@
+import { Transition } from "@headlessui/react";
+import Link from "next/link";
+import { useEffect, useState } from "react";
+
+import { MenuDropdown } from "~/components/control";
+import { Button } from "~/components/ui/button";
+import { useWindowSize } from "~/hooks";
+
+export const TransactionButtons = ({
+  open,
+  address,
+}: {
+  open: boolean;
+  address: string;
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  // Close the dropdown when the transaction details sidebar / modal is closed
+  useEffect(() => {
```

Comment:
Can we brainstorm ways to not use this useEffect to sync state? It opens us up to potential state sync issues/bugs

Perhaps we can pass the dropdown state and setter as props?